### PR TITLE
feat: Add battery reading wrapper

### PR DESCRIPTION
**Pull request by Fyustorm**
This change introduces hasBatterySensor, readBatteryLevel, and sendSensorReadCmd to handle battery level reading from devices. These functions allow checking for a battery sensor, reading the battery level, and sending a sensor read command to the device.

### CODE REVIEW

File: buttplug4j/src/main/java/io/github/blackspherefollower/buttplug4j/client/ButtplugClientDevice.java [hunk 1]
```diff
@@ -382,6 +383,115 @@ public final Future<ButtplugMessage> sendLinearCmd(final Pair<Double, Long>[] ve
...
+     */
+    public final Future<ButtplugMessage> sendSensorReadCmd(final int index, final String sensorType)
+            throws ButtplugDeviceException {
+
+        MessageAttributes attrs = getDeviceMessages().get("SensorReadCmd");
+        if (!(attrs instanceof SensorMessageAttributes)) {
+            throw new ButtplugDeviceException("Device doesn't support SensorReadCmd!");
+        }
+
+        final SensorReadCmd cmd = new SensorReadCmd(index, index);
```
Comment (blackspherefollower) [inline buttplug4j/src/main/java/io/github/blackspherefollower/buttplug4j/client/ButtplugClientDevice.java]: I think the first argument here should be `deviceIndex`

**Review by blackspherefollower — CHANGES_REQUESTED** [review]
I think one of the index's is off

File: buttplug4j/src/main/java/io/github/blackspherefollower/buttplug4j/client/ButtplugClientDevice.java [hunk 2]
```diff
@@ -382,6 +384,116 @@ public final Future<ButtplugMessage> sendLinearCmd(final Pair<Double, Long>[] ve
...
+     */
+    public final Future<ButtplugMessage> sendSensorReadCmd(final int sensorIndex, final String sensorType)
+            throws ButtplugDeviceException {
+
+        MessageAttributes attrs = getDeviceMessages().get("SensorReadCmd");
+        if (!(attrs instanceof SensorMessageAttributes)) {
+            throw new ButtplugDeviceException("Device doesn't support SensorReadCmd!");
+        }
+
+        final SensorReadCmd cmd = new SensorReadCmd(this.deviceIndex, ButtplugConsts.DEFAULT_MSG_ID);
```
Comment (blackspherefollower) [inline buttplug4j/src/main/java/io/github/blackspherefollower/buttplug4j/client/ButtplugClientDevice.java]: Ah, just realised the messageId has to be set from `client.getNextMsgId()`: the response will have the same messageId, which is what ties up the future handling.

**Review by blackspherefollower — CHANGES_REQUESTED** [review]

**Comment by Fyustorm** [comment]
In the last commit I changed to client.getNextMsgId() and I also added a timeout in the readBatteryLevel because in the case the device crashed or anything else it would hangs up indefinitely

**Pull request merged**